synapse/handlers/federation.py

```diff
@@ -2077,7 +2079,7 @@ async def _auth_and_persist_events(
         self,
         origin: str,
         room_id: str,
-        event_infos: Iterable[_NewEventInfo],
+        event_infos: Collection[_NewEventInfo],
```

Comment:
Is the length of `event_infos` necessary for this function? As far as I can see all usage only requires an Iterable.

Reply:
Yeah, iterables don't necessarily have a size functon.

Reply:
Oh, yes indeed.

Reply:
Huh, TIL!

Reply:
At least, I'm fairly sure iterables don't have a size and so you can't check for emptiness...